pkg/xds/generator/transparent_proxy_generator.go

```diff
@@ -25,26 +30,32 @@ func (_ TransparentProxyGenerator) Generate(ctx xds_context.Context, proxy *mode
 	}
 	sourceService := proxy.Dataplane.Spec.GetIdentifyingService()
 	meshName := ctx.Mesh.Resource.GetMeta().GetName()
-	const outboundName = "outbound:passthrough"
-	listener, err := envoy_listeners.NewListenerBuilder().
-		Configure(envoy_listeners.OutboundListener(outboundName, "0.0.0.0", redirectPortOutbound)).
-		Configure(envoy_listeners.FilterChain(envoy_listeners.NewFilterChainBuilder().
-			Configure(envoy_listeners.TcpProxy(outboundName, envoy_common.ClusterSubset{ClusterName: outboundName})).
-			Configure(envoy_listeners.NetworkAccessLog(meshName, envoy_listeners.TrafficDirectionUnspecified, sourceService, "external", proxy.Logs[mesh_core.PassThroughService], proxy)))).
-		Configure(envoy_listeners.OriginalDstForwarder()).
-		Build()
-	if err != nil {
-		return nil, errors.Wrapf(err, "could not generate listener: %s", outboundName)
-	}
-	cluster, err := envoy_clusters.NewClusterBuilder().
-		Configure(envoy_clusters.PassThroughCluster(outboundName)).
-		Build()
-	if err != nil {
-		return nil, errors.Wrapf(err, "could not generate cluster: %s", outboundName)
+
+	var outboundPassThroughCluster *envoy_api_v2.Cluster = nil
+	var outboundListener *envoy_api_v2.Listener = nil
+	var err error
+	if ctx.Mesh.Resource.Spec.GetNetworking().GetOutbound().GetPassthrough() {
```

Comment:
Let's do for now
```
if ctx.Mesh.Resource.Spec.GetNetworking().GetOutbound() == nill || ctx.Mesh.Resource.Spec.GetNetworking().GetOutbound().GetPassthrough() {
```
so the upgrade is more seamless

Reply:
> Can we maybe have E2E test for this? :)

Yes, can be part of the ExternalService E2E once it gets in.